src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -435,6 +424,22 @@ private void OnSymbolEnd(SymbolAnalysisContext symbolEndContext, bool hasInvalid
                 return;
             }
 
+            private static LocalizableString GetMessage(
+               DiagnosticDescriptor rule,
+               ISymbol member)
+            {
+                var messageFormat = rule.MessageFormat;
+                if (rule == s_removeUnreadMembersRule &&
+                    member is IMethodSymbol)
+                {
+                    // IDE0052 has a different message for method symbols.
+                    messageFormat = FeaturesResources.Private_method_0_can_be_removed_as_it_is_never_invoked;
+                }
```

Comment:
note.  my complaint here still stands (albeit it is minor).  It definitely is a smell to have a rule specify a format string, but then have the caller just override that in particular cases.

Reply:
Yep, but our diagnostic system provides no better solution here, other then reporting a diagnostic with separate ID here, which IMO is an overkill for this case.

Reply:
Your call :)